### PR TITLE
Compile with xcode12

### PR DESCRIPTION
**Pull request by JoeSSS**
Removing of arm64 arch is needed to compile the project with Xcode 12

### CODE REVIEW

File: AppStub/Info.plist [hunk 1]
```diff
@@ -15,7 +15,7 @@
 	<key>CFBundlePackageType</key>
 	<string>APPL</string>
 	<key>CFBundleShortVersionString</key>
-	<string>2.2.1</string>
+	<string>$(MARKETING_VERSION)</string>
```
Comment (jamuc) [inline AppStub/Info.plist:18]: Oh, fancy ...

File: DeviceAgent.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -3737,7 +3737,8 @@
 				CLANG_WARN_SUSPICIOUS_MOVE = YES;
 				CLANG_WARN_UNREACHABLE_CODE = YES;
 				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
-				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
+				CODE_SIGN_IDENTITY = "iPhone Distribution: Karl Krukow (FYD86LA7RE)";
```
Comment (jamuc) [inline DeviceAgent.xcodeproj/project.pbxproj:3740]: Not sure this should be here. Generic iPhone developer seems nicer. It depends on how this is managed, but fastlane actually can cover all of this.

Comment (jamuc) [reply DeviceAgent.xcodeproj/project.pbxproj:3740]: Also, who is Karl Krukow?

File: DeviceAgent.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -3809,6 +3812,7 @@
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
 				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
+				"LD_RUNPATH_SEARCH_PATHS[arch=*]" = "$(FRAMEWORK_SEARCH_PATHS)";
```
Comment (jamuc) [inline DeviceAgent.xcodeproj/project.pbxproj:3815]: nice

**Review by jamuc — COMMENTED** [review]
Looks good to me, the only thing I would change is the hardcoded Karl thing.

**Review by jamuc — APPROVED** [review]

**Comment by fr0l** [comment]
@JoeSSS @jamuc the changes in project file `DeviceAgent.xcodeproj/project.pbxproj` are not really needed to build this project with Xcode 12 (beta 5)

As it is outlined in [this comment for Carthage](https://github.com/Carthage/Carthage/issues/3019#issuecomment-665136323) they have similar problem and they have a workaround.

Here is an example how to build DeviceAgent with Xcode 12 without modifications of `DeviceAgent.xcodeproj/project.pbxproj`

```shell
xcconfig=$(mktemp /tmp/static.xcconfig.XXXXXX)

echo 'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200 = arm64 arm64e armv7 armv7s armv6 armv8' >> $xcconfig
echo 'EXCLUDED_ARCHS = $(inherited) $(EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_$(EFFECTIVE_PLATFORM_SUFFIX)__NATIVE_ARCH_64_BIT_$(NATIVE_ARCH_64_BIT)__XCODE_$(XCODE_VERSION_MAJOR))' >> $xcconfig

XCODE_XCCONFIG_FILE="$xcconfig" \
DEVELOPER_DIR=/Applications/Xcode-beta.app/Contents/Developer \
  xcodebuild \
  -workspace DeviceAgent.xcworkspace \
  -scheme "DeviceAgent" \
  -configuration "Debug" \
  clean \
  build \
  -sdk "iphonesimulator" \
  -destination "generic/platform=iOS Simulator"
```

Probably it's good to wait until Xcode 12 Golden Master is released in a month, and then decide about changes in `DeviceAgent.xcodeproj/project.pbxproj` regarding this workaround

File: DeviceAgent.xcodeproj/project.pbxproj (hunk 2 above)
Comment (jmoody) [reply DeviceAgent.xcodeproj/project.pbxproj:3740]: Karl is the original author of Calabash iOS and Android.

He founded a company in Denmark called Lesspainful which was bought by Xamarin. Xamarin was bought by Microsoft.

We continue to use his credentials for signing.

We cannot use a generic developer ID.

We don't want to use fastlane.

**Comment by JoeSSS** [comment]
> @JoeSSS @jamuc the changes in project file `DeviceAgent.xcodeproj/project.pbxproj` are not really needed to build DeviceAgent with Xcode 12 (beta 5) or it is too early to merge them 😃
> 
> Carhtage project has a discussion about similar issue ([Carthage/Carthage#3019](https://github.com/Carthage/Carthage/issues/3019))
> They have a temporary workaround as it is outlined in [this comment for Carthage](https://github.com/Carthage/Carthage/issues/3019#issuecomment-665136323)
> 
> Here is an example how to build DeviceAgent with Xcode 12 without modifications of `DeviceAgent.xcodeproj/project.pbxproj`
> 
> ```shell
> xcconfig=$(mktemp /tmp/static.xcconfig.XXXXXX)
> 
> echo 'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200 = arm64 arm64e armv7 armv7s armv6 armv8' >> $xcconfig
> echo 'EXCLUDED_ARCHS = $(inherited) $(EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_$(EFFECTIVE_PLATFORM_SUFFIX)__NATIVE_ARCH_64_BIT_$(NATIVE_ARCH_64_BIT)__XCODE_$(XCODE_VERSION_MAJOR))' >> $xcconfig
> 
> XCODE_XCCONFIG_FILE="$xcconfig" \
> DEVELOPER_DIR=/Applications/Xcode-beta.app/Contents/Developer \
>   xcodebuild \
>   -workspace DeviceAgent.xcworkspace \
>   -scheme "DeviceAgent" \
>   -configuration "Debug" \
>   clean \
>   build \
>   -sdk "iphonesimulator" \
>   -destination "generic/platform=iOS Simulator"
> ```
> 
> Probably it's good to wait until Xcode 12 Golden Master is released in a month, and then decide about changes in `DeviceAgent.xcodeproj/project.pbxproj` regarding this workaround.
> 
> Looks like this is a surprise from Apple while they are migrating to a new architecture on macOS. 😉

thanks for the hints! It also feels like a problem that can go away in the next beta or GM. I just wanted to have at least something in a branch, so we can build a version that allow us to test the app against iOS 14 already. Therefore this PR most likely will be closed

File: AppStub/Info.plist (hunk 1 above)
Comment (jmoody) [reply AppStub/Info.plist:18]: The bin/git-tag.sh script reads this Info.plist for the DeviceAgent tag.

https://github.com/calabash/DeviceAgent.iOS/blob/develop/bin/git-tag.sh#L5

Rollback this change or update the script to read the .xcodeproj file (hard) and update the documentation.

**Comment by jmoody** [comment]
This change breaks a run-loop test.

https://github.com/calabash/run_loop/blob/develop/spec/integration/device_agent/client_spec.rb#L17

It causes the DeviceAgent to be built for i386 targets only.

You can reproduce the problem:

```
COMMAND_LINE_BUILD=1 CLOBBER=1 \
  xcrun xcodebuild \
  -derivedDataPath ./DerivedData \
  -scheme AppStub \
  -workspace ./DeviceAgent.xcworkspace \
  -config Debug \
  -sdk iphonesimulator \
   ARCHS='i386 x86_64' \
   VALID_ARCHS='i386 x86_64' \
   ONLY_ACTIVE_ARCH=NO \
  -destination id=< use an x86 simulator UDID> \
   test
```

**Comment by fr0l** [comment]
@JoeSSS @jmoody maybe it is a good idea to wait until Xcode 12 official release, and follow resolution of similar issue Carthage/Carthage#3019 ?  😄

**Pull request closed without merging**